unstructured/ingest/connector/vectara.py

```diff
@@ -0,0 +1,226 @@
+        }
+
+        response = requests.request(method=http_method, url=url, headers=headers, params=params, data=json.dumps(data))
+        response.raise_for_status()
+        return response.json()
+
+    # get OAUth2 JWT token
+    def _get_jwt_token(self):
+        """Connect to the server and get a JWT token."""
+        token_endpoint = f"https://vectara-prod-{self.connector_config.customer_id}.auth.us-west-2.amazoncognito.com/oauth2/token"
+        headers = {
+            "Content-Type": "application/x-www-form-urlencoded",
+        }
+        data = {
+            "grant_type": "client_credentials", 
+            "client_id": self.connector_config.access_config.oauth_client_id, 
+            "client_secret": self.connector_config.access_config.oauth_secret
+        }
+
+        request_time = datetime.datetime.now().timestamp()
+        response = requests.request(method="POST", url=token_endpoint, headers=headers, data=data)
+        response_json = response.json()
+
+        self.jwt_token = response_json.get("access_token")
+        self.jwt_token_expires_ts = request_time + response_json.get("expires_in")
+        return self.jwt_token
+
+    @DestinationConnectionError.wrap
+    def check_connection(self):
+        """
+        Check the connection for Vectara and validate corpus exists.
+        - If more than one exists - then return a message
+        - If exactly one exists with this name - use it.
+        - If does not exist - create it.
+        """
+        try:
+            jwt_token = self._get_jwt_token()
+            if not jwt_token:
+                return "Unable to get JWT Token. Confirm your Client ID and Client Secret."
+
+            list_corpora_response = self._request(endpoint="list-corpora", data={"numResults": 100, "filter": self.connector_config.corpus_name})
+
+            possible_corpora_ids_names_map = {
+                corpus.get("id"): corpus.get("name")
+                for corpus in list_corpora_response.get("corpus")
+                if corpus.get("name") == self.connector_config.corpus_name
+            }
+
+            if len(possible_corpora_ids_names_map) > 1:
+                return f"Multiple Corpora exist with name {self.connector_config.corpus_name}"
+            if len(possible_corpora_ids_names_map) == 1:
+                self.connector_config.corpus_id = list(possible_corpora_ids_names_map.keys())[0]
+            else:
+                data = {
+                    "corpus": {
+                        "name": self.connector_config.corpus_name,
+                    }
+                }
+                create_corpus_response = self._request(endpoint="create-corpus", data=data)
+                self.connector_config.corpus_id = create_corpus_response.get("corpusId")
+
+        except Exception as e:
+            return str(e) + "\n" + "".join(traceback.TracebackException.from_exception(e).format())
+
+    def _get_headers(self) -> t.Dict[str, t.Any]:
```

Comment:
is this still used? Couldn't find it being called anywhere

Reply:
Yes, good point. Not used anymore after added _request(), will remove.